bin/logger

```diff
@@ -64,11 +64,53 @@ we redirect the aggregated output back to STDOUT (FD 1). You can also log the
 aggregated output to a single file.
 """
 
+import itertools
 import sys
 
 WIDTH = 20
 
 
+class Colours:
+    colours = itertools.cycle(
+        {
+            "cyan": "36",
+            "yellow": "33",
+            "green": "32",
+            "magenta": "35",
+            "red": "31",
+            "blue": "34",
+            "intense_cyan": "36;1",
+            "intense_yellow": "33;1",
+            "intense_green": "32;1",
+            "intense_magenta": "35;1",
+            "intense_red": "31;1",
+            "intense_blue": "34;1",
+        }.values()
+    )
+
+    def __init__(self):
+        self._process_colours = {}
+
+    def colour(self, process_name, s):
+        colour = self._process_colours.get(process_name)
+
+        if not colour:
+            colour = self._process_colours[process_name] = next(self.colours)
+
+        return self._colour_string(colour, s)
+
+    @staticmethod
+    def _ansi(code):
+        return "\033[{0}m".format(code)
+
+    @classmethod
+    def _colour_string(cls, colour, s):
+        return "{0}{1}{2}{3}".format(cls._ansi(0), cls._ansi(colour), s, cls._ansi(0))
```

Comment:
This is copied [from honcho](https://github.com/nickstenning/honcho/blob/master/honcho/colour.py) (including the colours and their order)